scripts/engineering/derive_phase44d_carry_aware_family_constant_surface.py

```
@@ -0,0 +1,580 @@
+    with path.open("r", encoding="utf-8") as handle:
+        rows = list(csv.DictReader(handle, delimiter="\t"))
+    if not rows:
+        raise SystemExit(f"no rows found in {path}")
+    return rows
+
+
+def parse_float(raw: str, *, label: str, path: Path) -> float:
+    try:
+        value = float(raw)
+    except ValueError as exc:
+        raise SystemExit(f"{label} must be numeric in {path}; got {raw!r}") from exc
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)